app/controllers/api/v4/questionnaires_controller.rb

```diff
@@ -10,9 +10,23 @@ def transform_to_response(questionnaire)
   def current_facility_records
     # TODO: Current implementation always responds with 1 JSON minimum. Reason:
     # process_token.last_updated_at has precision upto 3 milliseconds & is always lesser than updated_at.
+    dsl_version = params.require("dsl_version").to_f.truncate(2)
+
+    # For dsl_version 1.5, return all questionnaires from 1 to 1.5.
+    # De-duplicate multiple questionnaires of same type by choosing latest dsl_version.
     Questionnaire
       .for_sync
-      .where(dsl_version: params.require("dsl_version").to_i)
+      .select("*")
+      .joins("
+        INNER JOIN (
```

Comment:
Can this be written with a `DISTINCT ON` in postgres? Something like
```
select distinct on (questionnaire_type)
from questionnaires
where dsl_version between #{dsl_version.floor} AND #{dsl_version}
order by dsl_version

```

Reply:
Modified. Wow, distinct on really removed the complexity.